docs/guide/Microfrontend-JAMStack.md

```diff
@@ -1,243 +1,326 @@
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+    - run: yarn install --frozen-lockfile
+    - run: yarn lint
+    - run: yarn test
+    - run: yarn build
+#    - name: Slack Notification
+#      uses: rtCamp/action-slack-notify@v2.0.2
+#      env:
+#        SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+```
+
+### CDNs
+
+The resulting SPA is served by a CDN or multiple CDNs. Individual micro-frontends can live in different CDNs.
+
+### Kubernetes
+
+The resulting SPA can run and be served in one or more Kubernetes environments.
+
+### Host application
+
+The host application comes with a pre-configured setup out-of-the-box. It does not need any particular configuration different from a traditional SPA more than the Webpack 5 Module Federation configuration.
+It will be acting as the orchestrator, loading the remote micro frontends and providing them with app-wide functionality e.g. auth, RBAC, client side routing.
+
+There is virtually no limit on how the host can grow and how much can be extended.
+It is suggested however to centralize all the host-child communication and shared components in an external library so that both host and children have the same knowledge and integration won’t break.
+
+### Versioning microfrontends
+​The suggested approach is to build a registry where individual apps are registered. The registry would allow to set some configuration on each app and keep track of all the available versions.
+​
+It would then expose and api consumed by the host, providing informations around the available microfrontends, the versions and the artifact locations.
+​
+The registry would be administered by a trusted operator through a user interface; it would be the trusted operator responsibility to decide which version of each individual app would be made public and available to the host to load.
+It would also allow to easily test versions and rollback when necessary, all that without needing to rebuild and redeploy the apps.
+​
+**Note** The JS build artifacts created by Webpack do not include the version in the filename. It's could be necessary to upgrade the build in order to differentiate versions. A simpler approach that do not require to update the build configuration would be hosting the versions on different URLs.
+​
+### Upgrading the host
+​The host is pretty much self isolated and the only necessary thing to do proper versioning is to use the built in command `yarn version`. It will create a new git tag and increment the `package.json` version accordingly to how the command is used (interactive CLI).
+​
+### Upgrading the remotes
+​The remotes are self isolated and the only necessary thing to do proper versioning is to use the built in command `yarn version`. It will create a new git tag and increment the package.json version accordingly to how the command is used (interactive CLI).
```

Comment:
```suggestion
​The remotes are self-isolated and the only necessary thing to do proper versioning is to use the built-in command `yarn version`. It will create a new git tag and increment the `package.json` version according to how the command is used (interactive CLI).
```